plugin/pkg/admission/storage/persistentvolume/label/admission.go

```diff
@@ -292,15 +280,11 @@ func (l *persistentVolumeLabel) getAzureCloudProvider() (*azure.Cloud, error) {
 		if err != nil || cloudProvider == nil {
 			return nil, err
 		}
-		azureProvider, ok := cloudProvider.(*azure.Cloud)
-		if !ok {
-			// GetCloudProvider has gone very wrong
-			return nil, fmt.Errorf("error retrieving Azure cloud provider")
-		}
-		l.azureProvider = azureProvider
+
+		l.azurePVLabeler = cloudProvider.(cloudprovider.PVLabeler)
```

Comment:
do a conditional interface assertion so we can return an error instead of panicking?